meilisearch-http/src/snapshot.rs

```diff
@@ -0,0 +1,44 @@
+use std::path::{PathBuf, Path};
+use std::fs::{create_dir_all, File};
+use tar::Builder;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use std::thread;
+use std::time::{Duration};
+use tempfile::TempDir;
+use crate::error::Error;
+use crate::Data;
+
+
+pub fn create_snapshot(data: &Data, snapshot_path: &PathBuf) -> Result<(), Error> {
+    let tmp_dir = TempDir::new()?;
+
+    data.db.copy_and_compact_to_path(tmp_dir.path());
+
+    let f = File::create(snapshot_path)?;
+    let mut gz_encoder = GzEncoder::new(f, Compression::default());
+    {
+        let mut tar_encoder = Builder::new(&mut gz_encoder);
+
+        tar_encoder.append_dir_all(&data.db_path, tmp_dir.path())?;
+        tar_encoder.finish()?;
+    }
+    gz_encoder.finish()?;
+
+    Ok(())
+}
+
+pub fn schedule_snapshot(data: Data, snapshot_path: &PathBuf, time_gap_s: u64) -> Result<(), Error> {
+    if snapshot_path.file_name().is_none() { 
+        return Err(Error::Internal("invalid snapshot file path".to_string()));
+    }
+    if let Some(parent_dir) = snapshot_path.parent() { create_dir_all(parent_dir)?; }
+    let snapshot_path = snapshot_path.with_extension("tar.gz");
+    
+    thread::spawn(move || loop { 
+        thread::sleep(Duration::from_secs(time_gap_s));
+        create_snapshot(&data, &snapshot_path);
```

Comment:
Hello @MarinPostma @Kerollmops, here we have a possible error returned by `create_snapshot`, how can I log it properly?

Reply:
There is a log::error macro you can use to log an error